src/main/java/ch/wisv/payments/admin/products/ProductController.java

```diff
@@ -88,9 +89,48 @@ public String deleteProduct(@PathVariable int productId, RedirectAttributes redi
         return "redirect:/products";
     }
 
-    @PostMapping(value = "/addGroup")
+    @PostMapping(value = "/group")
     public String addProductGroup(@ModelAttribute @Validated ProductGroupRequest productGroupRequest) {
         productService.addProductGroup(productGroupRequest);
+        System.out.println("here");
+
+        return "redirect:/products";
+    }
+
+    @GetMapping(value = "/group/edit/{productGroupId}")
+    public String productGroupEdit(@PathVariable Integer productGroupId, Model model) {
+        model.addAttribute("committees", committeeService.getAllCommittees());
+
+        ProductGroup productGroup = productService.getProductGroupById(productGroupId);
+        ProductGroupRequest productGroupRequest = new ProductGroupRequest();
+
+        productGroupRequest.setId(productGroup.getId());
+        productGroupRequest.setName(productGroup.getName());
+        productGroupRequest.setCommitteeId(productGroup.getCommittee().getId());
+        productGroupRequest.setDescription(productGroup.getDescription());
+        productGroupRequest.setGroupLimit(productGroup.getGroupLimit());
+
+        model.addAttribute("productGroup", productGroupRequest);
+
+        return "editProductGroup";
+    }
+
+    @PostMapping(value = "/group/edit")
+    public String editProductGroup(@ModelAttribute @Validated ProductGroupRequest productGroupRequest, RedirectAttributes redirectAttributes) {
+        productService.editProductGroup(productGroupRequest);
+
+        redirectAttributes.addFlashAttribute("message", productGroupRequest.getName() + " successfully updated!");
+
+        return "redirect:/products";
+    }
+
+    @PostMapping(value = "/group/delete/{productGroupId}")
+    public String deleteProductGroup(@PathVariable int productGroupId, RedirectAttributes redirectAttributes) {
+        try {
+            productService.deleteProductGroup(productGroupId);
+        } catch (RuntimeException e) {
```

Comment:
Try to have that "redirect" to the current page, so you don't lose a filled in form.